### PR TITLE
Add compatibility with http-errors to jsonapi_exception_formatter

### LINKED ISSUE #1: Add compatibility with http-errors to jsonapi_exception_formatter
Currently the `jsonapi_exception_formatter` decorator normally handles `JsonApiException` exceptions only. Source:
```python
        try:
            return func(*args, **kwargs)
        except JsonApiException as e:
            return make_response(jsonify(jsonapi_errors([e.to_dict()])),
                                 e.status,
                                 headers)
        except Exception as e:
             ...
```
However, plugins can implement their own errors, which can be correlated with `JsonApiException` errors. This is necessary to avoid handling errors such as critical.
moreover, this library integrates with flask and should support its standard http-errors.
I suggest in the block `except Exception` to check the attribute code (or status_code) of the error and try to correlate it with the JSONAPI error. If a match is found, then do not log this error, but return the corresponding JSONAPI error.
Example:
```python
except Exception as e:
    http_ex = format_http_exception(e)
    if http_ex:
        return make_response(jsonify(jsonapi_errors([http_ex.to_dict()])),
                                 http_ex.status,
                                 headers)
```

### PR DESCRIPTION
**Pull request by pysalt**
The functionality described in task [https://github.com/AdCombo/flask-rest-jsonapi/issues/1](#1 ) is implemented

### CODE REVIEW

File: flask_rest_jsonapi/errors.py [hunk 1]
```diff
@@ -1,4 +1,12 @@
 """Helper to  format Api errors according to jsonapi specification"""
+from flask_rest_jsonapi.exceptions import BadRequest, ObjectNotFound, InvalidType, AccessDenied
+
+STATUS_MAP = {
+    400: BadRequest,
+    404: ObjectNotFound,
+    409: InvalidType,
+    403: AccessDenied,
```
Comment (Znbiz) [inline flask_rest_jsonapi/errors.py:9]: Need to add code 401

Comment (pysalt) [reply flask_rest_jsonapi/errors.py:9]: Lib doesnt have exception for this code. We can raise 401 as AccessDenied or create new exception like this `Unauthorized`

Comment (Znbiz) [reply flask_rest_jsonapi/errors.py:9]: Let's create a new exception

Comment (pysalt) [reply flask_rest_jsonapi/errors.py:9]: ok

File: flask_rest_jsonapi/exceptions.py [hunk 2]
```diff
@@ -98,6 +98,13 @@ class InvalidType(JsonApiException):
     status = '409'
 
 
+class Unauthorized(JsonApiException):
```
Comment (pysalt) [inline flask_rest_jsonapi/exceptions.py:101]: @Znbiz done

**Pull request merged**